kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/CollectionMatchersTest.kt

```diff
@@ -880,9 +881,7 @@ class CollectionMatchersTest : WordSpec() {
                   { it shouldBeGreaterThan 7 }
                )
             }.message shouldBe """
-               Expected each element to pass its assertion, but found issues at indexes: [2]
```

Comment:
the number of failed assertion seems meaningful, is there a reason it's removed?

Reply:
2 is not a number here, but an index. The logic behind this removal is that the failure just one and it is identified by the stacktrace. It can be changed, but the API of `MatcherResult` is quite hard to work with, and I don't know the overall code so well, so I'm not sure if I wouldn't break anything.

Reply:
> 2 is not a number here, but an index. The logic behind this removal is that the failure just one and it is identified by the stacktrace. It can be changed, but the API of `MatcherResult` is quite hard to work with, and I don't know the overall code so well, so I'm not sure if I wouldn't break anything.

I agree that adding stacktrace is an improvement and it should definitely help in some situations, but IMO there are cases when it's just easier and faster to go by the index of the failed predicate. So can we please add back the index of the failed predicate.

Reply:
Ok, changed it that way.